plugins/node/opentelemetry-instrumentation-mysql2/src/utils.ts

```diff
@@ -107,22 +108,41 @@ function getJDBCString(
 export function getDbStatement(
   query: string | Query | QueryOptions,
   format?: formatType,
-  values?: any[]
+  values?: any[],
+  maskStatement: boolean = true,
+  maskStatementHook: MySQL2InstrumentationQueryMaskingHook = defaultMaskingHook
 ): string {
-  if (!format) {
-    return typeof query === 'string' ? query : query.sql;
-  }
-  if (typeof query === 'string') {
-    return values ? format(query, values) : query;
-  } else {
-    // According to https://github.com/mysqljs/mysql#performing-queries
-    // The values argument will override the values in the option object.
-    return values || (query as QueryOptions).values
-      ? format(query.sql, values || (query as QueryOptions).values)
-      : query.sql;
+  let formattedQuery: string;
+  try {
+    if (maskStatement) {
+      formattedQuery = typeof query === 'string' ? query : query.sql;
+      return maskStatementHook(formattedQuery);
+    }
+
+    if (!format) {
+      formattedQuery = typeof query === 'string' ? query : query.sql;
+    } else if (typeof query === 'string') {
+      formattedQuery = values ? format(query, values) : query;
+    } else {
+      // According to https://github.com/mysqljs/mysql#performing-queries
+      // The values argument will override the values in the option object.
+      formattedQuery =
+        values || (query as QueryOptions).values
+          ? format(query.sql, values || (query as QueryOptions).values)
+          : query.sql;
+    }
+    return formattedQuery;
+  } catch (e) {
+    return 'Could not determine the query due to an error in masking or formatting';
   }
```

Comment:
I find this series of overlapping conditionals to be really hard to follow, and I feel like it could definitely be simplified. Something like this maybe ? I'm also not sure there are any reasons to keep the formatting logic at all apart from backwards compatibility since it does the exact opposite of what the spec asks for.
```suggestion
  const [querySql, queryValues] = typeof query === 'string'
    ? [query, values]
    // According to https://github.com/mysqljs/mysql#performing-queries
    // The values argument will override the values in the option object.
    : [query.sql, values || query.values];
  try {
    if (maskStatement) {
      return maskStatementHook(querySql);
    } else if (format && values) {
      return format(querySql, queryValues);
    } else {
      return querySql;
    }
  } catch (e) {
    return 'Could not determine the query due to an error in masking or formatting';
  }
```

Reply:
I adjusted it, but added the `hasValues` Typecheck as well, since Typescript was complaining otherwise (Query don't has the properties values).

The formatting is actual needed, since its replacing the placeholders with the real values. If someone is setting maskStatement to 'false' I assume, the intent is, that all queries are pushed unmasked into the trace, not only already unmasked Queries. Therefore we need to resolve the parameterized Queries as well. I can remove it, if you think, that this is not beneficial at all.